src/geometry/manifold/times_cont_mdiff.lean

```diff
@@ -1287,6 +1324,60 @@ lemma smooth_within_at_proj
     (proj I M) s p :=
 basic_smooth_bundle_core.smooth_within_at_proj _
 
+variables (I M)
+/-- The zero section of the tangent bundle -/
+def zero_section : M → tangent_bundle I M := λ x, ⟨x, 0⟩
+variables {I M}
+
+lemma smooth_zero_section : smooth I I.tangent (zero_section I M) :=
+begin
+  apply basic_smooth_bundle_core.smooth_const_section (tangent_bundle_core I M) 0,
+  assume i j x hx,
+  simp only [tangent_bundle_core, continuous_linear_map.map_zero] with mfld_simps
+end
+
+/-- The derivative of the zero section of the tangent bundle maps `⟨x, v⟩` to `⟨⟨x, 0⟩, ⟨v, 0⟩⟩`.
+
+Note that this statement relies on the choice of the coordinate system we use on the tangent bundle.
+However, there is a canonical one at a point of the zero section (since there is a canonical
+horizontal space there, the tangent space to the zero section, in addition to the canonical vertical
+space which is the kernel of the derivative of the projection), and this is the one that comes
+from our definitions, so this statement is not as crazy as it may seem. In general, avoid using it
+though, since coordinate-free concepts are always better. -/
+lemma tangent_map_tangent_bundle_pure (p : tangent_bundle I M) :
+  tangent_map I I.tangent (tangent_bundle.zero_section I M) p = ⟨⟨p.1, 0⟩, ⟨p.2, 0⟩⟩ :=
```

Comment:
Is the following co-ordinate invariant statement (a) well-formed, (b) equivalent: the tangent map of the zero section (of TM over M) is the zero section (of T(TM) over TM)?

Reply:
Re (b): Won't the zero section of TTM over TM take a point W to (W,0)? In this case the action of the tangent map maps it to an element whose second component is nonzero (`⟨p.2, 0⟩`)

Reply:
Oh ... I see! :(

Personally, I find it sad to add a non-coordinate-free statement to the library, especially at this early stage.  This lemma (as opposed to the obviously-useful `smooth_zero_section`) was just a puzzle that @kodyvajjha invented, not needed for any project, is that correct?  Maybe it could move to become an exercise in the LFTCM project?

Someday mathlib will have Riemannian geometry, with a splitting of `T(TM)`, and it will be possible to make sense of a statement like this.

Reply:
Sorry, I should have put this in context probably. I wanted to investigate the tangent functor monad: https://arxiv.org/pdf/1401.0940.pdf
The hope was that one can use this monad to simplify certain constructions/state things cleaner.

Since the library is very mature I believe one can endow this `T` with a monad structure. This lemma which @sgouezel proved came up when I was trying to formalize theorem 2.1.1 in the above paper (Incidentally, that paper has a coordinate-free restating of the unit laws in theorem 2.1.1).

Reply:
As I try to explain in the docstring, this lemma is in fact coordinate-free in the sense that there is a canonical splitting of T_{(x,0)} (TM) into a direct sum, and this is the splitting that comes naturally out of the definition. I certainly don't want to include non-canonical statements in mathlib.

Reply:
I see.  What is the minimum setup to express this canonical vertical projection (along the zero section) in a coordinate-free way?  I guess you can do it if you have defined bundle maps and pullbacks of bundles, because then this vertical projection is a bundle(-over-M) map from $z^*T(TM)$ to $TM$; the tangent map is a bundle(-over-M) map from $TM$ to $z^*T(TM)$, and this lemma is the statement that their composition is the zero bundle map.

Reply:
> What is the minimum setup to express this canonical vertical projection (along the zero section) in a coordinate-free way? I guess you can do it if you have defined bundle maps and pullbacks of bundles, 

(obviously that's not for this PR, but could you write it as a TODO in the docstring for when that theory has been developed?)

Reply:
The minimum setup, I think, would be to say that at the zero section there is a direct sum decomposition of T_{(x,0)}(TM) into a direct sum im (derivative of the zero section) \oplus ker (derivative of the projection), and that the image of the derivative of the zero section applied to v is (v, 0) in this decomposition. 

Another version is to say that, for any connection on the bundle and the resulting decomposition of the tangent space to the tangent bundle, then the image of v is (v, 0) (as along the zero section the decomposition coming from the connection is in fact independent of the connection, and coincides with the aforementioned decomposition).